hudi-flink/src/main/java/org/apache/hudi/sink/partitioner/BucketAssignFunction.java

```diff
@@ -207,13 +226,18 @@ private HoodieRecordLocation getNewRecordLocation(String partitionPath) {
     final HoodieRecordLocation location;
     switch (bucketInfo.getBucketType()) {
       case INSERT:
```

Comment:
Is there any code path that we may get `APPEND_ONLY` value from the `bucketInfo.getBucketType()`?